docs/developer-docs/backend/arguments.mdx

```diff
@@ -0,0 +1,262 @@
+
+```motoko
+type Day = {#sun; #mon; #tue; #wed; #thu; #fri; #sat};
+
+public query func get_text(d : Day) : Text {
+  switch d {
+     case (#sun) "Sunday";
+     case (#mon) "Monday";
+     case (#tue) "Tuesday";
+     case (#wed) "Wednesday";
+     case (#thu) "Thursday";
+     case (#fri) "Friday";
+     case (#sat) "Saturday";
+   };
+};
+```
+For additional examples, refer to the [Type variant reference in Motoko](https://internetcomputer.org/docs/current/motoko/main/overview/#variants).
+
+To learn more about variants in Motoko, refer to Kai Peacock's blog on [using Motoko variants](https://kyle-peacock.com/blog/motoko/using-variants/#:~:text=Motoko%20variants%20are%20a%20cool,storing%2C%20similar%20to%20an%20enum.). 
+
+</TabItem>
+
+<TabItem value="rust" label="Rust">
+
+```rust
+type Day = variant {
+    Sun;
+    Mon;
+    Tue;
+    Wed;
+    Thu;
+    Fri;
+    Sat;
+}
+
+[query]
+fn get_text(d: Day) -> &'static str {
+    match d {
+        Day::Sun => "Sunday",
+        Day::Mon => "Monday",
+        Day::Tue => "Tuesday",
+        Day::Wed => "Wednesday",
+        Day::Thu => "Thursday",
+        Day::Fri => "Friday",
+        Day::Sat => "Saturday",
+    }
+}
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="Typescript">
+
+```typescript
+import { Canister,  query, Variant } from 'azle';
+
+const Day = Variant({
+    Sun;
+    Mon;
+    Tue;
+    Wed;
+    Thu;
+    Fri;
+    Sat;
+});
+
+export default Canister({
+    get_text: query([Day], Day, (d)) => {
+        switch (d) {
+            case Day.Sun:
+                return "Sunday";
+            case Day.Mon:
+                return "Monday";
+            case Day.Tue:
+                return "Tuesday";
+            case Day.Wed:
+                return "Wednesday";
+            case Day.Thu:
+                return "Thursday";
+            case Day.Fri:
+                return "Friday";
+            case Day.Sat:
+                return "Saturday";
+    }
+}
+});
+```
+To learn more about variants in Typescript via Azle, refer to [the Azle book reference on variants](https://demergent-labs.github.io/azle/reference/candid/variant.html?highlight=variant#variant).
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```python
+from kybra import nat32, Variant, query
+
+class Day(Variant, total=False):
+    Sun: str
+    Mon: str
+    Tue: str
+    Wed: str
+    Thu: str
+    Fri: str
+    Sat: str
+
+@query
+def get_text(d: Day) -> str:
+    if d == Day.Sun:
+        return "Sunday"
+    elif d == Day.Mon:
+        return "Monday"
+    elif d == Day.Tue:
+        return "Tuesday"
+    elif d == Day.Wed:
+        return "Wednesday"
+    elif d == Day.Thu:
+        return "Thursday"
+    elif d == Day.Fri:
+        return "Friday"
+    elif d == Day.Sat:
+        return "Saturday"
+```
+
+To learn more about variants in Python in Kybra, refer to [the Kybra book reference on variants](https://demergent-labs.github.io/kybra/reference/candid/variant.html).
```

Comment:
It looks like Markdown does not believe that the previous code block has closed.

Reply:
Yes, the rendering seems correct here when I do a preview. :(

Reply:
Jessie agreed that we can merge now and see if the page does not render properly.